integration-tests/tests/common/mod.rs

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! This crate contains the common code for integration tests
+
+#![deny(clippy::all)]
+#![deny(missing_docs)]
+
+use std::path::PathBuf;
+use std::process::Command;
+use std::time::Duration;
+use wait_timeout::ChildExt;
+
+#[cfg_attr(not(any(has_sgx, has_sev)), ignore)]
+pub fn keep_under_test() -> &'static str {
+    let mut result = "";
+    if cfg!(has_sgx) {
+        result = "enarx-keep-sgx";
+    }
+    if cfg!(has_sev) {
+        result = "enarx-keep-sev";
+    }
+    result
+}
+
+pub struct IntegrationTest {
+    pub wksp_root: PathBuf,
+    pub keep: PathBuf,
+    pub payload: PathBuf,
+    pub shim: PathBuf,
+}
+
+impl IntegrationTest {
+    pub fn new(bin: &str) -> Self {
+        let enarx_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let enarx_dir = enarx_dir.parent().unwrap();
+        let keep_type = keep_under_test();
+        Self {
+            wksp_root: enarx_dir.to_path_buf(),
+            keep: find_filepath(
+                enarx_dir
+                    .join(format!("{}/target/", keep_type))
+                    .to_path_buf(),
+                format!("{}", keep_type).as_str(),
+            )
+            .unwrap(),
+            payload: find_filepath(
+                enarx_dir
+                    .join("integration-tests/target/x86_64-unknown-linux-musl/")
+                    .to_path_buf(),
+                bin,
+            )
+            .unwrap(),
+            shim: find_filepath(
+                enarx_dir
+                    .join(format!(
+                        "{}-shim/target/x86_64-unknown-linux-musl/",
+                        keep_type
+                    ))
+                    .to_path_buf(),
+                format!("{}-shim", keep_type).as_str(),
+            )
+            .unwrap(),
+        }
+    }
+    pub fn run_test(
+        self: Self,
+        timeout: Option<usize>,
+        exit_status: Option<i32>,
+        _stdin: Option<&str>,
+        _stdout: Option<&str>,
+        _stderr: Option<&str>,
+    ) -> () {
+        let mut cmd = Command::new(self.keep)
+            .current_dir(self.wksp_root)
+            .arg("--code")
+            .arg(self.payload)
+            .arg("--shim")
+            .arg(self.shim)
+            .spawn()
+            .expect("failed to run the test");
+        let num_sec = match timeout {
+            Some(num_sec) => num_sec,
+            None => 5,
+        };
+        let exit_status = match exit_status {
+            Some(exit_status) => exit_status,
+            None => 0,
+        };
+        let ecode = match cmd
+            .wait_timeout(Duration::from_secs(num_sec as u64))
+            .unwrap()
+        {
+            Some(status) => status.code(),
+            None => {
+                cmd.kill().unwrap();
+                panic!("killed by watchdog!");
+            }
+        };
+        assert_eq!(exit_status, ecode.unwrap());
```

Comment:
```suggestion
        let seconds = Duration::from_secs(timeout.unwrap_or(5) as u64);
        let exit_status = exit_status.unwrap_or(0);

        let mut cmd = Command::new(self.keep)
            .current_dir(self.wksp_root)
            .arg("--code")
            .arg(self.payload)
            .arg("--shim")
            .arg(self.shim)
            .spawn()
            .expect("failed to run the test");
        
        let ecode = match cmd.wait_timeout(seconds).unwrap() {
            Some(status) => status.code(),
            None => {
                cmd.kill().unwrap();
                panic!("killed by watchdog!");
            }
        };

        assert_eq!(exit_status, ecode.unwrap());
```

1. I used vertical whitespace to group related blocks of code.
2. I replaced `match` with `.unwrap_or()` where applicable.
3. I moved `Duration::from_secs()` earlier, which removed unnecessary wrapping.
4. I moved input argument processing to the top of the function.

It still might be possible to simplify the remaining match statement if I looked more closely. But it may not be worth it.

Notice that the groups of code are now:
1. Process input arguments.
2. Prepare the command for execution.
3. Execute the command.
4. Evaluate the results.

This should be very easy to read now.

Reply:
Note that I incuded `.unwrap_r()` even though it isn't needed anymore. I just wanted to let you know that it exists.